plugins/UM3NetworkPrinting/src/Models.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+from typing import Optional
+
+
+class BaseModel:
```

Comment:
We have a basic model system that provides more features in https://github.com/Ultimaker/stardust-websocket_protocol/blob/95a4e1ff/protocol/models/BaseModel.py
I prefer to reuse that than to create a new one

Reply:
I discussed this with Marijn and we decided it was not a good idea to bring that whole (part of the) code base into Cura at the moment (as no data modeling is done at all yet in this repo).

Reply:
I think it's actually better to use named tuples in this case; https://stackoverflow.com/questions/34269772/type-hints-in-namedtuple